src/test/java/spoon/test/ctBlock/TestCtBlock.java

```diff
@@ -83,10 +84,10 @@ public void testAddStatementInBlock() {
 
 		assertEquals(5, block.getStatements().size());
 
-		assertEquals(originalBlock.getStatement(0), block.getStatement(0));
+		assertEquals((CtStatement) originalBlock.getStatement(0), block.getStatement(0));
```

Comment:
Is this cast really necessary?

Reply:
![image](https://user-images.githubusercontent.com/25300639/149567426-ccdb1b49-aa92-4a28-89e7-3973015f75e6.png) Looks like otherwise the compiler has some type problems.

Reply:
There's gotta be something wrong here, how in the world would the overload resolve to `Short`? I'll have a look and see if my compiler can give us a second opinion, but it'll have to wait until tomorrow :)

Reply:
Argh, okay now I get it. `CtStatementList.getStatement()` is typed with a type parameter that's only used in the return type, see: #1846

It can possibly be solved, but it's just not worth the effort.